docs/src/main/asciidoc/secretmanager.adoc

```diff
@@ -0,0 +1,83 @@
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+  compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-starter-secretmanager'
+}
+----
+
+=== Secret Manager Property Source
+
+The Spring Cloud GCP integration for Google Cloud Secret Manager enables you to use Secret Manager as a bootstrap property source.
+
+This feature allows you to automatically load your GCP project's secrets as properties into the application context during the https://cloud.spring.io/spring-cloud-commons/reference/html/#the-bootstrap-application-context[Bootstrap Phase], which refers to the initial phase when a Spring application is being loaded.
+
+NOTE: To begin using this feature, you must set the setting `spring.cloud.gcp.secretmanager.bootstrap.enabled` to **true**, this feature disabled by default.
+
+Spring Cloud GCP will load the **latest** version of each secret into the application context.
+
+All secrets will be loaded into the application environment using their `secretId` as the property name.
+For example, if your secret's id is `my-secret` then it will be accessible as a property using the name `my-secret`.
+If you would like to append a prefix string to all property names imported from Secret Manager, you may use the `spring.cloud.gcp.secretmanager.secret-name-prefix` setting described below.
+
+Spring Cloud GCP Secret Manager offers several configuration properties to customize the behavior.
+
+|===
+| Name | Description | Required | Default value
+| `spring.cloud.gcp.secretmanager.bootstrap.enabled` | Enables loading secrets from Secret Manager as a bootstrap property source. Set this to **true** to enable the feature. | No | `false`
+| `spring.cloud.gcp.secretmanager.credentials.location` | OAuth2 credentials for authenticating with the Google Cloud Secret Manager API. | No | By default, infers credentials from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
+| `spring.cloud.gcp.secretmanager.credentials.encoded-key` | Base64-encoded contents of OAuth2 account private key for authenticating with the Google Cloud Storage API. | No | By default, infers credentials from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
+| `spring.cloud.gcp.secretmanager.project-id` | The GCP Project used to access Secret Manager API. | No | By default, infers the project from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
+| `spring.cloud.gcp.secretmanager.secret-name-prefix` | A prefix string to prepend to the property names of secrets read from Secret Manager | No | "" (empty string)
+|===
+
+=== Secret Manager Template
+
+The `SecretManagerTemplate` class simplifies operations of creating, updating, and reading secrets.
+
+To begin using this class, you may autowire an instance of the class using `@Autowired` after adding the starter dependency to your project.
```

Comment:
Should we just vaguely say "inject" instead of "autowire... using autowired" despite showing an example of it?
In production code, autowire would not be commonly used.

Reply:
Done.